setup.py

```diff
@@ -82,9 +96,76 @@
         'Programming Language :: Python :: 3 :: Only',
         'Topic :: Scientific/Engineering']
 }
-# do not compile external modules on darwin
-if platform.system() in ["Windows", "Linux"]:
-    setup_kwargs["ext_modules"] = [fim_module]
 
```

Comment:
Here, a main branching could be used to check for a general flag to stop compilations. This would become handy in case extra modules are added and the user would not need to name them all. Inside the main `if`, each individual extension is then treated. Something in the lines of `--no-compile` or `--no-extensions`.

Reply:
Do you mean similar to this?
```python
if '--no-extensions' not in sys.argv:
    if platform.system() in ["Windows", "Linux"]:
        extensions = []
        options = {"--nomodule1": module1, "--nofim": fim_module}
        for flag in options:
            if flag not in sys.argv:
                    extensions.append(options[flag])

        setup_kwargs["ext_modules"] = extensions
    ...

```

Reply:
Yes. Perhaps more Pythonic with a list comprehension


```
if '--no-extensions' not in sys.argv:
    if platform.system() in ["Windows", "Linux"]:
        options = {"--nomodule1": module1, "--nofim": fim_module}
        extensions = [module for flag, module in options.items() if flag not in sys.argv]
        setup_kwargs["ext_modules"] = extensions
```

Reply:
I agree, this looks nice !

Reply:
see: [a1f0c57](https://github.com/NeuralEnsemble/elephant/pull/494/commits/a1f0c575ccdc90e6528f6fa845c84968dfc4fa82)